### PR TITLE
Add pre-commit to dev-requirements, run on existing files.

### PR DESCRIPTION
**Pull request by mlists**
Trailing whitespace doesn't seem to show correctly on github, the files are all showing EOF newlines locally.

### CODE REVIEW

**Comment by auscompgeek** [comment]
I don't think it makes sense to install `pre-commit` in the same virtualenv as robot code. That's kinda like installing virtualenv in a manually managed virtualenv to create another virtualenv.

pre-commit has a few dependencies that won't end up on the robot, and it'd be preferable for the tests in CI to run without those dependencies, to ensure nobody accidentally uses them.

You can use `pipx` or your system package manager to install `pre-commit`.

**Comment by mlists** [comment]
Fair call, I was trying to reduce the number of manual installs the new recruits will need to do. I will add instructions to install from pipx to the readme instead.

File: README.md [hunk 1]
```diff
@@ -11,11 +11,19 @@ pip install -r requirements-dev.txt
 ```
 
 ### Setup pre-commit
+Pre-commit is setup to automatically run formatters and linters when you commit. It is suggested to install pre-commit via pipx to keep it seperate from robot code dependencies. First install pipx:
+```
+pip install pipx
+```
```
Comment (auscompgeek) [inline README.md:17]: I'd rather point to the [pipx install docs](https://pipx.pypa.io/stable/installation/). A plain `pip install` is almost always the wrong thing to do.

File: README.md [hunk 2]
```diff
@@ -11,11 +11,19 @@ pip install -r requirements-dev.txt
...
-Pre-commit is setup to automatically run formatters and linters when you commit.
+Install pre-commit with pipx:
+```
+pipx install pre-commit
+```
 
+Setup the pre-commit hooks to run on commit:
 ```
-pre-commit install
+pipx run pre-commit install
```
Comment (auscompgeek) [inline README.md:26]: If `pipx` is installed correctly, `pipx install` will put things on $PATH.

```suggestion
pre-commit install
```

**Review by auscompgeek — APPROVED** [review]
Thanks!

**Pull request merged**